code/modules/mob/living/living.dm

```diff
@@ -1167,3 +1167,35 @@
 	C.take_organ_damage(damage)
 	C.KnockDown(3 SECONDS)
 	C.visible_message("<span class='danger'>[C] crashes into [src], knocking them both over!</span>", "<span class='userdanger'>You violently crash into [src]!</span>")
+
+/mob/living/throw_impact(atom/hit_atom, throwingdatum)
+	. = ..()
+	if(!has_status_effect(STATUS_EFFECT_CHARGING) || has_status_effect(STATUS_EFFECT_IMPACT_IMMUNE))
+		return
+
+	var/hit_something = FALSE
+	if(ismovable(hit_atom))
+		var/atom/movable/AM = hit_atom
+		var/atom/throw_target = get_edge_target_turf(AM, dir)
+		if(!AM.anchored || ismecha(AM))
+			AM.throw_at(throw_target, 5, 12, src)
```

Comment:
is this supposed to be `!ismecha(AM)` ? You may wanna check move_force here instead

Reply:
mechas are actually anchored

Reply:
I did copy this from the gargantua stuff, I think it is intended that mechs get thrown back?